apps/site/src/pages/settings/billing-settings-panel/card-details-panel-page.tsx

```diff
@@ -0,0 +1,200 @@
+      const matchingPaymentMethod = paymentMethods.find(
+        ({ id }) => id === router.query.id,
+      );
+
+      if (!matchingPaymentMethod) {
+        void router.push(paymentMethodsPanelPageAsPath);
+      }
+
+      return matchingPaymentMethod;
+    }
+  }, [router, paymentMethods]);
+
+  const currentBillingDetails = paymentMethod?.billing_details;
+
+  useEffect(() => {
+    if (currentBillingDetails) {
+      const { name, address } = currentBillingDetails;
+      setUpdatedBillingDetails({
+        name: name ?? undefined,
+        address: {
+          line1: address?.line1 ?? undefined,
+          line2: address?.line2 ?? undefined,
+          city: address?.city ?? undefined,
+          state: address?.state ?? undefined,
+          postal_code: address?.postal_code ?? undefined,
+          country: address?.country ?? undefined,
+        },
+      });
+    }
+  }, [currentBillingDetails]);
+
+  const handleSubmit = useCallback(
+    async (event: FormEvent) => {
+      event.preventDefault();
+
+      if (!paymentMethod || !updatedBillingDetails) {
+        return;
+      }
+
+      setUpdatingBillingDetails(true);
+
+      await internalApi.updatePaymentMethod(paymentMethod.id, {
+        updatedBillingDetails,
+      });
+
+      setUpdatingBillingDetails(false);
+    },
+    [paymentMethod, updatedBillingDetails],
+  );
+
+  const stripeElementsOptions = useMemo<StripeElementsOptions>(() => ({}), []);
+
+  return (
+    <>
+      <Typography
+        variant="bpBodyCopy"
+        sx={{ color: ({ palette }) => palette.gray[90], marginBottom: 4 }}
+      >
+        Change the details on file associated with your{" "}
+        {paymentMethod?.card?.brand &&
+        isKnownStripeCardBrand(paymentMethod.card.brand)
+          ? `${stripeCardBrandToHumanReadable[paymentMethod?.card?.brand]} `
+          : ""}{" "}
+        card
+      </Typography>
+      <Typography
+        variant="bpBodyCopy"
+        sx={{
+          textTransform: "uppercase",
+          color: ({ palette }) => palette.gray[90],
+        }}
+      >
+        <strong>Card on file</strong>
+      </Typography>
+      <Box marginBottom={4}>
+        {paymentMethod ? <PaymentMethod paymentMethod={paymentMethod} /> : null}
+      </Box>
+      <Typography
+        variant="bpBodyCopy"
+        sx={{
+          textTransform: "uppercase",
+          color: ({ palette }) => palette.gray[90],
+          marginBottom: 1,
+        }}
+      >
+        <strong>Billing address</strong>
+      </Typography>
+      <Elements stripe={stripePromise} options={stripeElementsOptions}>
+        {currentBillingDetails ? (
+          <Box maxWidth={420} component="form" onSubmit={handleSubmit}>
+            <AddressElement
+              options={{
+                mode: "billing",
+                defaultValues: {
+                  name: currentBillingDetails.name,
+                  address: {
+                    line1: currentBillingDetails.address?.line1 ?? undefined,
```

Comment:
These `defaultValues` should set the fields to their current value, right?
I am getting empty fields whenever I try to edit an address that I already entered before.

Reply:
> These defaultValues should set the fields to their current value, right?

Yes

> I am getting empty fields whenever I try to edit an address that I already entered before.

That's strange, I'm unable to reproduce this locally. Perhaps something is going wrong during the save process? Could you provide me with reproduction steps for this?

Reply:
Maybe something is missing in my environment, LMK if you want to make a call to figure this out tomorrow. 

![repro](https://user-images.githubusercontent.com/37453800/217118763-0c65e106-85cc-4358-a981-7982e835992e.gif)

Reply:
Let's pick this up in slack, I'm still having trouble reproducing this behaviour

Reply:
Ah I figured out the reproduction steps, and have fixed the bug in https://github.com/blockprotocol/blockprotocol/pull/933/commits/92f616d2416cd0a86f395f0fdc954161a63a52d6. Essentially the local `paymentMethods` weren't refetched after the payment details of a payment method were updated, which meant the form was populated with outdated fields if there was no hard-page refresh.

Reply:
Thank you @benwerner01, this is now working for me